lib/src/stack_zone_specification.dart

```diff
@@ -208,7 +208,9 @@ class StackZoneSpecification {
     try {
       return f();
     } catch (e, stackTrace) {
-      _chains[stackTrace] = node;
+      if (_chains[stackTrace] == null) {
+        _chains[stackTrace] = node;
+      }
```

Comment:
Nit: `??=` would make this more compact.

It would also be good to add a comment here mentioning that we can see the same stack trace multiple times if it's rethrown through guarded callbacks, and that the innermost chain will have the most information so it should take precedence.